src/components/cylc/tree/index.js

```diff
@@ -0,0 +1,148 @@
+        children: [],
+        id: parsedFamilyId.cyclePoint,
+        state: ''
+      }
+      lookup.set(parsedFamilyId.cyclePoint, cyclePointNode)
+      // a cycle point must go directly under the workflow
+      workflow.children.push(cyclePointNode)
+    }
+
+    // skip the root family, which has null as its first parent
+    if (parent === null) {
+      continue
+    }
+    if (!lookup.get(familyProxy.id)) {
+      // create family proxy node for the tree
+      Object.assign(familyProxy, {
+        __type: 'family',
+        name: parsedFamilyId.family,
+        children: []
+      })
+      lookup.set(familyProxy.id, familyProxy)
+      // we should not add to the parent just yet, as it could a) not exist, or b) be the root family
+    }
+
+    const parsedParentId = parseFamilyId(parent.id)
+    if (parsedParentId.family === 'root') {
+      // if the parent is root, we use the cyclepoint as the parent
+      lookup.get(parsedFamilyId.cyclePoint).children.push(lookup.get(familyProxy.id))
+    } else {
+      if (!lookup.get(parent.id)) {
+        // construct preliminary parent node if not root
+        Object.assign(parent, {
+          __type: 'family',
+          name: parsedParentId.family,
+          children: []
+        })
+        lookup.set(parent.id, parent)
+      }
+      // attach the family node to its parent (cycle point, or another family, except root family)
+      lookup.get(parent.id).children.push(lookup.get(familyProxy.id))
+    }
+  }
+
+  workflow.children.sort((cyclepoint, anotherCyclepoint) => {
+    return cyclepoint.id.localeCompare(anotherCyclepoint.id)
+  })
+
+  // simply iterate through tasks, creating the nodes, then attach them to their parents using the lookup map
+  for (const taskProxy of workflow.taskProxies) {
+    Object.assign(taskProxy, {
+      __type: 'task',
+      name: taskProxy.task.name,
+      children: [],
+      expanded: false
+    })
+    const parsedTaskProxyId = parseFamilyId(taskProxy.firstParent.id)
+    if (parsedTaskProxyId.family === 'root') {
+      // if the parent is root, we must instead attach this node to the cyclepoint!
+      lookup.get(parsedTaskProxyId.cyclePoint).children.push(taskProxy)
+    } else {
+      lookup.get(taskProxy.firstParent.id).children.push(taskProxy)
+    }
+    for (const job of taskProxy.jobs) {
+      Object.assign(job, {
+        __type: 'job',
+        name: `#${job.submitNum}`,
+        latestMessage: taskProxy.latestMessage
+      })
+      taskProxy.children.push(job)
+    }
+    // calculate task progress if necessary/possible
+    if (STATES_WITH_PROGRESS.includes(taskProxy.state) && taskProxy.jobs.length > 0) {
+      // the graphql query is expected to have jobs sorted by submit_num, e.g.:
+      // `jobs(sort: { keys: ["submit_num"], reverse:true })`
+      const latestJob = taskProxy.jobs[0]
+      if (Object.hasOwnProperty.call(latestJob, 'startedTime')) {
+        const startedTime = Date.parse(latestJob.startedTime)
```

Comment:
The JavaScript `Date` function can only parse ISO8601 which uses the hypen/colon syntax (which is optional in ISO8601):

```
> new Date('20000101T0000Z')
Invalid Date
> new Date('2000-01-01T00:00Z')
2000-01-01T00:00:00.000Z
```

It also doesn't seem to be able to handle ISO8601 time zones:

```
> new Date('2000-01-01T00:00')
2000-01-01T00:00:00.000Z
> new Date('2000-01-01T00:00Z')
2000-01-01T00:00:00.000Z
> new Date('2000-01-01T00:00+01')
Invalid Date
```

So long as `startedTime` etc, are always in the extended (`-`+`:`) ISO8601 format in the UTC time zone we are good (what happens when you remove `[cylc]UTC mode = True`?), must make a note to ensure this interface is preserved.

Reply:
I feel like we had a similar discussion recently-ish, I think @hjoliver commented in a PR where I had a note about maybe using moment.js or another library for handling the date returned from GraphQL, but that for the moment it appeared to work fine with the vanilla JS functions.

>So long as startedTime etc, are always in the extended (-+:) ISO8601 format in the UTC time zone we are good (what happens when you remove [cylc]UTC mode = True?), must make a note to ensure this interface is preserved.

+1 should we add a note here to the code, to README.md, create an issue in this repo? WDYT?

Reply:
> maybe using moment.js

ISO8601 in UTC is definitely the way we should be doing things so we shouldn't need the parsing / TZ functionality of `moment.js`.

> +1 should we add a note here to the code, to README.md, create an issue in this repo? WDYT?

Dunno, perhaps a small issue on Cylc Flow for someone to check dates passed over the API are preserved in this format.